quinn-proto/src/connection/mod.rs

```diff
@@ -889,23 +889,30 @@ where
         Some(id)
     }
 
-    /// Read from the given recv stream, in undefined order
+    /// Read from the given recv stream
     ///
-    /// While stream data is typically processed by applications in-order, unordered reads improve
-    /// performance when packet loss occurs and data cannot be retransmitted before the flow control
-    /// window is filled. When in-order delivery is required, the sibling `read()` or `read_chunk[s]()`
-    /// methods should be used.
+    /// `max_length` limits the maximum size of the returned `Bytes` value; passing `usize::MAX`
+    /// will yield the best performance. Chunk sizes should be no larger than the UDP MTU (in the
+    /// 1500 bytes range, unless IPv6 jumbograms are supported). `ordered` will make sure the
+    /// returned chunk's offset will have an offset exactly equal to the previously returned offset
+    /// plus the previously returned bytes' length.
```

Comment:
I don't think we should make any guarantees (let alone fuzzy ones) about chunk sizes.

Reply:
Removed that part of the text.